docs/06-contributors/999-rfcs/000-stories/story-15.md

```diff
@@ -0,0 +1,171 @@
+# User Story 15 - Task List with Api Gateway and Redis
+
+> **Status**: Draft, Expected released on 2023/03/16
+
+
+The following code is an inital implementation of TaskList with api gateway and a redis db 
+
+## New language and SDK features it introduces
+
+- [x] bring untyped
+  - [x] bring external npm package (axios)
+  - [x] bring an internal nodejs stdlib (RegEx)
+  - [x] How does untyped works with numeric operations?
+  - [x] Do we cast untyped? 
+- [x] enum & duration that can be included inside json
+- [x] It leverages setting explicit permissions (using the `this.inflight` API, described [here](https://github.com/winglang/wing/pull/1610))
+- [x] Using interface 
+- [x] Using await when calling a untyped function that returns a promise
+- [ ] bring cdktf
+- [ ] use redis instead of bucket
+- [ ] code that updates estimation and duration from REST post command
+
+## Discussion topics
+- `get` vs `on_get` - Should the `cloud.api` API have the `on_` prefix to match `cloud.bucket` API and also to allow calling
+the api get/post/delete/put commands (`api.get(url)` vs `api.on_get(path, inflight ())`
+- Routing params - I have used express's synatx for dynamic parts of the path for the api gateway `get("/task/:id"`, 
+but then I noticed that [aws](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-create-api-step-by-step.html) 
+uses the `{id}` syntax. What should be the the right syntax?
+- Bring internal nodejs types - how do we bring somethink that we can't require, like `RegEx`
+- From anything to Json - can I cast something from bring untyped to Json? 
+- Calling await 
+
+## Code 
+```ts (wing)
+
+bring cloud;
+
+// TODO discuss how we bring untyped something like RegEx from JavaScript 
+// PLACEHOLER for bringing something from Javascript stdlib
+bring untyped js;
+
+// prerequisite: npm install axios
+// PALCEHOLDER for bringing some external module
+bring untyped "axios" as axios; 
```

Comment:
@eladb 

Once we've added `bring untyped js;`
It makes me wonder why aren't we allowing:

```ts
let axios = js.require("axios");
```
instead of 
```ts
bring untyped "axios" as axios; 
```

Reply:
And then we should wonder if we even need `bring untyped` as a mechanism and the `untyped` keyword vs having some dedicated way to being `js`

Reply:
`require` is a node.js thing, not a JS thing, and it's also not a "real function". I wouldn't conflate module imports and function calls (that's a big mistake node didthe JS community is still trying to recover from using `import`).